### PR TITLE
Scoring System

### PR DESCRIPTION
**Pull request by bloier**
- Add PointsController with "points" endpoint and getPoints & reset methods
- Add PointsService with calculate points method
- Update findRandomLetters method in WordflowRepo to actually give randomised letters from a list
- Add usePoints as custom hook to get points from new endpoint in backend
- Update GamePage with points and add layout for it

### CODE REVIEW

File: backend/src/main/java/de/wordsmith/backend/service/PointsService.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    private final WordflowRepository wordflowRepository;
+
+    public PointsService(WordflowRepository wordflowRepository) {
+        this.wordflowRepository = wordflowRepository;
+    }
+
+    public void calculateAndAddPoints(Word newWord) {
+        int increasedPoints = wordflowRepository.getPoints() + 10;
+        System.out.println(increasedPoints);
```
Comment (not-enough-ram) [inline backend/src/main/java/de/wordsmith/backend/service/PointsService.java:18]: please remove console output :)

File: backend/src/main/java/de/wordsmith/backend/service/WordflowService.java [hunk 2]
```diff
@@ -22,11 +22,27 @@ public Letters findRandomLetters(){
...
     public void addWordToWordsList(Word newWord) {
         wordflowRepository.addWordToWordsList(newWord);
+        calculateAndAddPoints(newWord);
     }
 
-    public boolean doesWordExist (String wordText) {
-        return ponsApiService.doesWordExist(wordText);
+    private void calculateAndAddPoints(Word newWord) {
+        int increasedPoints = wordflowRepository.getPoints() + 10;
+        System.out.println(increasedPoints);
```
Comment (not-enough-ram) [inline backend/src/main/java/de/wordsmith/backend/service/WordflowService.java:36]: this one as well :)

File: frontend/src/hooks/usePoints.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import axios from "axios";
+import {useState} from "react";
+
+export default function usePoints() {
+    const [points, setPoints] = useState(null);
+
+    function getPoints() {
```
Comment (not-enough-ram) [inline frontend/src/hooks/usePoints.js:7]: maybe have a look at react queries they make request handling a bit easier to manage by providing useful helper functions

File: frontend/src/pages/GamePage.js [hunk 4]
```diff
@@ -13,13 +14,17 @@ export default function GamePage() {
...
 
   useEffect(() => {
     if (letters) {
       setWord(letters);
+      getPoints()
     }
-  }, [letters]);
+  }, [letters, points]);
 
+  /** Click on button "Spielen" sets variable "isRunning" to true and activates timer with 30 seconds countdown **/
```
Comment (not-enough-ram) [inline frontend/src/pages/GamePage.js:27]: I guess coaches would tell us to not use any comments in code XD

**Review by not-enough-ram — APPROVED** [review]
lgtm 👍

File: frontend/src/pages/GamePage.js (hunk 4 above)
Comment (bloier) [reply frontend/src/pages/GamePage.js:27]: Yes but my comments will always get updated ;)

**Pull request merged**